internal/pkg/block/block_test.go

```diff
@@ -56,7 +56,7 @@ func TestTriangleEncoding(t *testing.T) {
 	}
 	t.Run("encoding block with zero fields works", func(t *testing.T) {
 		testRoundTrip(t, &blk.Block{
-			BlockSig:        crypto.Signature{Type: crypto.SigTypeSecp256k1, Data: []byte{}},
+			BlockSig:        &crypto.Signature{Type: crypto.SigTypeSecp256k1, Data: []byte{}},
 			BLSAggregateSig: crypto.Signature{Type: crypto.SigTypeBLS, Data: []byte{}},
```

Comment:
Should this also be a pointer? What happens if there are no BLS messages? 

I guess that's what this test is exercising, and Lotus also uses a non-pointer here.

Reply:
> Should this also be a pointer?

That *would* be more consistent with the block sig convention and save us bytes I think.  We'd need to make some implementation changes but probably worth it.  At the very least we need to specify whether this can be serialized as nil or not.  

> Lotus also uses a non-pointer here

Yeah  I'm viewing this as fine tuned spec work that's probably valuable in the long run but not immediately necessary since we seem to be bytewise consistent across impls.

Reply:
> save us bytes
a byte